### PR TITLE
Statically link to Rustls instead of using OpenSSL

### PR DESCRIPTION
**Pull request by charlespierce**
Info
-----
* Implements the changes described in https://github.com/volta-cli/rfcs/pull/47
* As discussed in the RFC, while using a statically-linked TLS implementation reduces our need to build for the unique CentOS flavor of OpenSSL, we still need to use an older version of Linux to link against an older `glibc` version.
* However, the build tools on CentOS 6 (the baseline for the `glibc` support that we've maintained) are too old to support building `ring` (a dependency of `rustls`).
* To work around that, I discovered a docker image for [Scientific Linux CERN 6](https://linux-archive.web.cern.ch/scientific6/), which is based off of RHEL 6 (the same as CentOS).
    * The CERN distro includes its own mirror of the package repositories (in fact, we are already using the cern mirrors for CentOS 6 in our CI right now)
    * It also includes the ability to install more recent build tools, which allows us to compile `ring`

Changes
-----
* Updated the dependency on `attohttpc` to enable the use of `rustls` instead of `native-tls` (OpenSSL on Linux)
* Created a `Dockerfile` that builds off of Scientific Linux CERN 6, installing updated build tools
* Updated the `release` CI workflow to build a single Linux version using the above Docker container.
* Updated the install script to look for the new single version in the case of Linux

Tested
-----
* Confirmed locally using this Docker image that we are able to compile and run Volta
* Verified that the single Linux build artifact from CI runs correctly works on old CentOS Linux, recent Ubuntu, and latest Ubuntu (3 different OpenSSL versions).

Notes
-----
* This PR changes the file name for the `linux` build artifact to be `volta-{version}-linux.tar.gz`, without the `openssl-X` suffix. This means we probably will need to roll it out in stages:
    * Pin the redirect for `get.volta.sh` to a commit hash, so that it isn't affected when this is merged. (Done in https://github.com/volta-cli/get/pull/7) (_Completed_)
    * Merge this and prepare the release (along with any other features that we might want to include)
    * Test by fetching the new script directly to confirm that it works as expected.
    * Cut the release, then un-pin the `get.volta.sh` redirect so that it points at the updated script.
* We'll also need to update the "Installing Old Versions" section of the [Installers](https://docs.volta.sh/advanced/installers) docs to include another special-cased version of the install script, for people who want to install versions prior to the install script change.

### CODE REVIEW

**Comment by alvarlagerlof** [comment]
Would this mean that you wouldn't need openssl to install volta?

**Comment by charlespierce** [comment]
@alvarlagerlof Yes, exactly! That's actually one of the main motivations for this suggested change (see the linked RFC): Dealing with all of the different OpenSSL versions—trying to detect the appropriate one, fetch the correct binary, etc.—is a major pain point in our Linux installer. This would simplify all of that by statically linking to a Rust-based TLS/SSL implementation and avoiding all of the headaches that come with dynamically-linked OpenSSL.

**Comment by ZauberNerd** [comment]
@charlespierce I guess this PR would also allow https://github.com/volta-cli/volta/issues/1049 to move forward, right?
I'm using volta on an EC2 with the Graviton2 ARM CPU and at the moment have to compile it myself, but would much prefer volta to release prebuilt binaries.

**Comment by charlespierce** [comment]
@ZauberNerd Yes, I believe this would allow that change to move forward as well, since it would remove the OpenSSL issues that ran into. The shift to Scientific Linux CERN 6 may also help alleviate the toolchain issues.

**Comment by ZauberNerd** [comment]
@charlespierce that's awesome! Is there something I could do to help get this PR out of the draft state?

**Comment by charlespierce** [comment]
@ZauberNerd I don't think there's anything left on this PR directly (other than resolving the merge conflicts). I'm leaving this in draft for now mostly to make sure that we don't accidentally merge it :smile: There's some infrastructure work we'll need to do before merging to make sure we don't break existing installs while we work towards the actual release of this change.

**Comment by chriskrycho** [comment]
@charlespierce what (if anything) can I do to unblock this over this coming week? Would love to help get it landed!

**Comment by charlespierce** [comment]
@chriskrycho I think the only thing remaining is landing this PR (and then some testing when we cut the release). My initial thoughts were being a little wary of version churn with the in-progress pnpm implementation, but I think there's enough significant changes between this and Yarn 3 support, plus there are some follow-ups needed for pnpm, so I'd lean towards getting this merged (along with any smaller outstanding PRs), and then cutting it all as version 1.1.

I'll update this to resolve the merge conflicts and also put out a PR to clean up some additional warnings we seem to be getting.

**Comment by chriskrycho** [comment]
That seems reasonable to me! Let's do it!

File: dev/unix/volta-install.sh
Comment (chriskrycho) [inline dev/unix/volta-install.sh:134]: Just double-checking—this isn't a regression, is it? I know we had had folks *interested* in doing Arm builds on Linux, and AFAIK we had never *shipped* those but also hadn’t done things which prevented people from doing it themselves… but don't remember if that was the extent of it?

**Review by chriskrycho — APPROVED** [review]
Other than the one question, this looks good to me.

File: dev/unix/volta-install.sh
Comment (charlespierce) [reply dev/unix/volta-install.sh:134]: Correct, though double-checking we don't actually need that. There is already a `check_architecture` function that prints an error message if the wrong architecture is used.

There was some work to try to get ARM builds for Linux, but I believe it stalled on CI build issues (that I think the approach in this PR should unblock)

**Comment by charlespierce** [comment]
itshappening.gif

**Comment by chriskrycho** [comment]
That's precisely the plan—this and Yarn 3 support will be the headliner features! (With, hopefully, a 1.2 with pnpm support landing *relatively* soon after that. 🤞🏼)

File: dev/unix/volta-install.sh
Comment (vladfrangu) [reply dev/unix/volta-install.sh:134]: Hey! Would like to add that I this this *does* cause a regression for arm64 users of Linux trying to build and install volta locally (conversation from https://canary.discord.com/channels/413753499835301908/413753499835301917/1026109337313222736 lower) 😅 
Basically this change prevents `./dev/unix/volta-install.sh --release` from being ran on arm64.

Comment (charlespierce) [reply dev/unix/volta-install.sh:134]: Ah, thanks for the call-out @vladfrangu! Will get that reverted (since as noted above it isn’t actually necessary anyway).

**Pull request merged**
